SimpleNameTagPlugin/src/main/java/eu/cloudnetservice/simplenametag/SimpleNameTagSystem.java

```diff
@@ -0,0 +1,167 @@
+            Bukkit.broadcast("In order to prevent issues, the name (+ tagID) of the group " + permissionGroup.getName() + " was temporarily shortened to 16 characters!",
+                    "cloudnet.notify");
+            Bukkit.broadcast("Please fix this issue by changing the name of the group in your perms.yml", "cloudnet.notify");
+        }
+        Team team = all.getScoreboard().getTeam(teamName);
+        if (team == null) {
+            team = all.getScoreboard().registerNewTeam(teamName);
+        }
+
+        if (permissionGroup.getPrefix().length() > 16) {
+            permissionGroup.setPrefix(permissionGroup.getPrefix().substring(0, 16));
+            CloudAPI.getInstance()
+                    .dispatchConsoleMessage("In order to prevent issues, the prefix of the group " + permissionGroup.getName() + " was temporarily shortened to 16 characters!");
+            CloudAPI.getInstance().dispatchConsoleMessage("Please fix this issue by changing the prefix in your perms.yml");
+            Bukkit.broadcast("In order to prevent issues, the prefix of the group " + permissionGroup.getName() + " was temporarily shortened to 16 characters!",
+                    "cloudnet.notify");
+            Bukkit.broadcast("Please fix this issue by changing the prefix in your perms.yml", "cloudnet.notify");
+        }
+        if (permissionGroup.getSuffix().length() > 16) {
+            permissionGroup.setSuffix(permissionGroup.getSuffix().substring(0, 16));
+            CloudAPI.getInstance()
+                    .dispatchConsoleMessage("In order to prevent issues, the suffix of the group " + permissionGroup.getName() + " was temporarily shortened to 16 characters!");
+            CloudAPI.getInstance().dispatchConsoleMessage("Please fix this issue by changing the suffix in your perms.yml");
+            Bukkit.broadcast("In order to prevent issues, the suffix of the group " + permissionGroup.getName() + " was temporarily shortened to 16 characters!",
+                    "cloudnet.notify");
+            Bukkit.broadcast("Please fix this issue by changing the suffix in your perms.yml", "cloudnet.notify");
+        }
+
+        try {
+            Method setColor = team.getClass().getDeclaredMethod("setColor", ChatColor.class);
+            setColor.setAccessible(true);
+            if (permissionGroup.getColor().length() != 0) {
+                setColor.invoke(team, ChatColor.getByChar(permissionGroup.getColor().replaceAll("&", "").replaceAll("§", "")));
+            } else {
+                setColor.invoke(team, ChatColor.getByChar(ChatColor.getLastColors(permissionGroup.getPrefix().replace('&', '§'))
+                        .replaceAll("&", "")
+                        .replaceAll("§", "")));
+            }
+        } catch (NoSuchMethodException ignored) {
+        } catch (IllegalAccessException | InvocationTargetException e) {
+            e.printStackTrace();
+        }
```

Comment:
Is it necessary to use reflections for 1.13+ plugins in this case?

Reply:
Yes is very necessary. Or we have problems with 1.13 backwards compatibility